### PR TITLE
New filter for lost password link in wp-login.php

### PR DESCRIPTION
**Pull request by dream-encode**
Trac ticket: https://core.trac.wordpress.org/ticket/55388

### CODE REVIEW

**Comment by SergeyBiryukov** [comment]
Hi there, thanks for the PR! I have some notes:

* `lost_password_url` sounds pretty similar to the existing `lostpassword_url` filter in `wp_lostpassword_url()`. Since the new hook filters the whole link and not just the URL, I think `lost_password_html_link` would be a better name, for consistency with the `login_site_html_link` filter added in [r50117](https://core.trac.wordpress.org/changeset/50117). I would personally drop `html`, but would prefer consistency here 🙂 
* There is exactly the same link below the registration form, it could also use the new filter.

02beb4d addresses these suggestions.

**Review by costdev — APPROVED** [review]
LGTM! :+1:

**Review by anton-vlasenko — APPROVED** [review]
LGTM!
Please see the test report here: https://core.trac.wordpress.org/ticket/55388#comment:10

File: src/wp-login.php [hunk 1]
```diff
@@ -1103,8 +1103,16 @@ function wp_login_viewport_meta() {
 
 		<p id="nav">
 			<a href="<?php echo esc_url( wp_login_url() ); ?>"><?php _e( 'Log in' ); ?></a>
-				<?php echo esc_html( $login_link_separator ); ?>
-			<a href="<?php echo esc_url( wp_lostpassword_url() ); ?>"><?php _e( 'Lost your password?' ); ?></a>
+			<?php
+
+			echo esc_html( $login_link_separator );
```
Comment (hellofromtonya) [inline src/wp-login.php:1108]: Why `echo esc_html()` instead of `esc_html_e()`?

```suggestion
			esc_html_e( $login_link_separator );
```

Comment (hellofromtonya) [reply src/wp-login.php:1108]: Oh I see. It's repeated multiple times within the same file. So here, it's for consistency 👍

**Review by hellofromtonya — APPROVED** [review]
LGTM 👍 Marking for `commit`.

**Comment by dream-encode** [comment]
Merged into core in https://core.trac.wordpress.org/changeset/54206

**Pull request closed without merging**